docs/how_to_write_custom_backend.rst

```diff
@@ -0,0 +1,193 @@
+
+        def __init__(self, **options):
+            super().__init__(**options)
+
+            # Lower case it just to be sure
+            options = {key.lower(): value for key, value in options.items()}
+            self._key = options.get("key", None)
+            self._secret = options.get("secret", None)
+            self._from = options.get("from", None)
+
+            # Create a Nexmo Client object
+            self.client = nexmo.Client(key=self._key, secret=self._secret)
+
+Initialize your class constructor with ``options`` dictionary which contains all the settings specific to your service defined in ``settings.py``. We have fetched each setting from ``options`` in above piece of code. Apart from it, we have created a client for our service by providing it the necessary settings.
+
+4. Override ``send_sms`` method of ``BaseBackend`` class to implement functionality for sending sms. It must have two positional parameters ``number`` and ``message`` respectively.
+
+.. code-block:: python
+
+    ...
+
+    def __init__(self, **options):
+        super().__init__(**options)
+
+        # Lower case it just to be sure
+        options = {key.lower(): value for key, value in options.items()}
+        self._key = options.get("key", None)
+        self._secret = options.get("secret", None)
+        self._from = options.get("from", None)
+
+        # Create a Nexmo Client object
+        self.client = nexmo.Client(key=self._key, secret=self._secret)
+
+    def send_sms(self, number, message):
+        # Implement your service's send sms functionality
+        self.client.send_message({
+            'from': self._from,
+            'to': number,
+            'text': message,
+        })
+
+5. For sending bulk messages, you must override ``send_bulk_sms`` method of ``BaseBackend`` class. It must have two positional parameters ``numbers`` and ``message`` respectively.
+
+.. code-block:: python
+
+    ...
+
+    def send_sms(self, number, message):
+        # Implement your service's send sms functionality
+        self.client.send_message({
+            'from': self._from,
+            'to': number,
+            'text': message,
+        })
+
+    def send_bulk_sms(self, numbers, message):
+        for number in numbers:
+            self.send_sms(self, number=number, message=message)
+
+How to create custom Sandbox Service
+------------------------------------
+
+The above steps will remain same if you wish to create a sandbox utility for your service. We'll create a new class with keeping above steps in mind. Apart from it, we will need to override a few more methods and tweak our ``__init__`` method a bit.
+
+1. Create a custom sandbox class for your service. Let's say ``NexmoSandboxBackend``. Again, it must inherit from ``BaseBackend``.
+
+2. Define your class's constructor same as above. Just fetch ``SANDBOX_TOKEN`` from the settings as well. Also, override ``send_sms`` and ``send_bulk_sms`` for your service same as above.
```

Comment:
```suggestion
2. The constructor for your sandbox environment can get the ``SANDBOX_TOKEN`` from settings as shown. This enables you to keep the token constant for testing purposes. You can then override ``send_sms`` and ``send_bulk_sms`` for the service as done in case of actual custom backend written.
```